### DISCUSSION TITLE
Short-hand syntax for single op effect-handlers

### DISCUSSION THREAD

**Discussion by magnus-madsen** [flix/flix#7094] [Ideas]
Given:

```scala
eff Ask {
    pub def ask(): String
}

eff Say {
    pub def say(s: String): Unit
}

def greeting(): Unit \ {Ask, Say} = 
    let name = do Ask.ask();
    do Say.say("Hello Mr. ${name}")

def main(): Unit \ IO = 
    try {
       greeting() 
    } with Ask {
        def ask(k) = k("Bond, James Bond")
    } with Say {
        def say(s, k) = { println(s); k() }
    }
```

The proposal, in the style of Effekt, is to allow one to write:

```scala
def main(): Unit \ IO = 
    try {
       greeting() 
    } with Ask { k -> k("Bond, James Bond") }
      with Say { (s, k) -> { println(s); k() } }
```

(The placement of braces is up for discussion.)

**Comment by magnus-madsen** [comment]
@mlutze @JonathanStarup CC

**Comment by JonathanStarup** [comment]
Hmm it saves `def` and writing the name, doesn't seem like much for the complexity. If you write your first example without newlines after `{` I don't think its as convincing. This feels close to first class handlers, maybe we should think about that? that might lead to simple syntax.
